xsd-core/src/main/java/io/ballerina/xsd/core/Utils.java

```diff
@@ -0,0 +1,133 @@
+                               VERTICAL_BAR + CLOSE_BRACES + SEMICOLON);
+        String extendedValue = nodes.get(element)
+                .toString().replace(type + WHITESPACE + CONTENT_FIELD + SEMICOLON, fields);
+        ModuleMemberDeclarationNode moduleNode = NodeParser.parseModuleMemberDeclaration(extendedValue);
+        nodes.put(element, moduleNode);
+    }
+
+    static void processSingleTypeElements(Map<String, ModuleMemberDeclarationNode> nodes,
+                                          String element, String type, String[] tokens) {
+        String token = (!nodes.containsKey(type)) || nodes.get(type).toString().contains(XSDVisitorImpl.ENUM)
+                ? STRING : tokens[tokens.length - 2];
+        String rootElement = nodes.get(element).toString().replace(type + WHITESPACE + CONTENT_FIELD,
+                token + WHITESPACE + CONTENT_FIELD);
+        ModuleMemberDeclarationNode moduleNode = NodeParser.parseModuleMemberDeclaration(rootElement);
+        nodes.put(element, moduleNode);
+    }
+
+    public static String extractSubstring(String baseString, String startToken, String endToken) {
+        if (!baseString.contains(startToken)) {
+            return baseString.split(WHITESPACE)[baseString.split(WHITESPACE).length - 2] +
+                    WHITESPACE + CONTENT_FIELD + SEMICOLON;
+        }
+        int startIndex = baseString.indexOf(startToken) + startToken.length();
+        int endIndex = baseString.indexOf(endToken, startIndex);
+        return baseString.substring(startIndex, endIndex);
+    }
+
+    public static NodeList<ImportDeclarationNode> getImportDeclarations(XSDVisitor xsdVisitor) throws Exception {
+        Collection<ImportDeclarationNode> imports = new ArrayList<>();
+        for (String module : xsdVisitor.getImports()) {
+            ImportDeclarationNode node = NodeParser.parseImportDeclaration(module);
+            if (node.hasDiagnostics()) {
+                throw new Exception(XSDToRecord.INVALID_IMPORTS_ERROR);
+            }
+            imports.add(node);
+        }
+        return AbstractNodeFactory.createNodeList(imports);
+    }
+
+    public static String formatModuleParts(ModulePartNode modulePartNode) throws FormatterException {
+        ForceFormattingOptions forceFormattingOptions = ForceFormattingOptions.builder()
+                .setForceFormatRecordFields(true).build();
+        FormattingOptions formattingOptions = FormattingOptions.builder()
+                .setForceFormattingOptions(forceFormattingOptions).build();
+        return Formatter.format(modulePartNode.syntaxTree(), formattingOptions).toSourceCode();
+    }
+
+    public static String resolveNameConflicts(String name, Map<?, ?> nodes) {
+        String baseName = name;
+        int counter = 1;
+        while (nodes.containsKey(name)) {
+            name = baseName + counter;
+            counter++;
+        }
+        return name;
+    }
+
+    public static String extractTypeName(String[] values) {
```

Comment:
Since the utils are used within the same package, can we make all the utils package-private?